sklearn/datasets/tests/test_lfw.py

```diff
@@ -36,44 +29,46 @@
 ]
 
 
-def setup_module():
-    """Test fixture run once and common to all tests of this module"""
-    Image = pytest.importorskip("PIL.Image")
+@pytest.fixture(scope="module")
+def mock_empty_data_home(tmp_path_factory):
+    data_dir = tmp_path_factory.mktemp("scikit_learn_empty_test")
 
-    global SCIKIT_LEARN_DATA, SCIKIT_LEARN_EMPTY_DATA, LFW_HOME
+    yield data_dir
```

Comment:
I think you can now use `return`

Reply:
I can but, I'd rather always use `yield` so I don't have to think "when should I use yield, when should I use return" ...

FWIW Pytest doc also recommends using yield fixtures: https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended